app/src/main/resources/application.properties

```diff
@@ -110,3 +110,49 @@ quarkus.http.cors=true
 # Disable OpenAPI class scanning
 mp.openapi.scan.disable=true
 mp.openapi.servers=/api
+
+# Auth
+
+%dev.quarkus.oidc.auth-server-url=http://localhost:8090/auth/realms/registry
+%dev.quarkus.oidc.client-id=registry-api
+
+%dev.quarkus.http.auth.permission.user.roles-allowed=registry-user
+%dev.quarkus.http.auth.permission.user.paths=/api/*,/public/*,/css/*,/js/*,/robots.txt
+%dev.quarkus.http.auth.permission.user.policy=permit
+%dev.quarkus.http.auth.permission.user.methods=GET,HEAD
+
+%dev.quarkus.http.auth.permission.user-deny.roles-allowed=registry-user
+%dev.quarkus.http.auth.permission.user-deny.paths=/api/*,/public/*,/css/*,/js/*,/robots.txt
+%dev.quarkus.http.auth.permission.user-deny.policy=deny
+%dev.quarkus.http.auth.permission.user-deny.methods=PUT,POST,DELETE
+
+%dev.quarkus.http.auth.permission.admin.roles-allowed=registry-admin
+%dev.quarkus.http.auth.permission.admin.paths=/api/*,/public/*,/css/*,/js/*,/robots.txt
+%dev.quarkus.http.auth.permission.admin.policy=permit
+%dev.quarkus.http.auth.permission.admin.methods=GET,HEAD,PUT,POST,DELETE
```

Comment:
So does this mean that a `registry-user` can only read and a `registry-admin` role is required to create/edit anything?

Reply:
Yes, that's what this means. You have more info [here](https://quarkus.io/guides/security).

Reply:
OK just making sure I understand.  We'll need to figure out the actual use-cases/functionality we want to support.  I'll set up a call with Keith et al to discuss this.

Reply:
As my understand so registry-user and registry-admin are a role of user from Keycloak right? I try to clone this branch and run in my local (using already my keycloak server), but I got the 403 when try to create/update/delete even my user already have registry-admin.

Reply:
Your description is correct as I understand it.  The Keycloak user you authenticated as has the `registry-admin` role?  I assume you've double-checked that?  :) :)  If so, then I'll have to test locally and see what I get.  I haven't tried this yet.

Reply:
Yeah, the keycloak user I'm using to log in has the registry-admin role. I think you should try to test your locally if you have time. 
and in this page: https://quarkus.io/guides/all-config
I don't see any config like this: 
quarkus.http.auth.permission.user.roles-allowed
Just have
quarkus.http.auth.policy."role-policy".roles-allowed
![image](https://user-images.githubusercontent.com/42263729/90534940-8f1ab600-e1a4-11ea-8c9b-45fdbe10c2ce.png)

Reply:
This is an error. I'll fix it.

Reply:
I saw the commit you put on the yesterday but your config is not working as we expected, I had checked and it's not working for user. With your config the admin has full permission but the user can't do anything even GET.

I had some change, and it's work correct, so you can test and update your config.
----------------------------

%prod.quarkus.http.auth.policy.read-policy.roles-allowed=user,admin
%prod.quarkus.http.auth.permission.user.paths=/api/*,/public/*,/css/*,/js/*,/robots.txt
%prod.quarkus.http.auth.permission.user.policy=read-policy
%prod.quarkus.http.auth.permission.user.methods=GET,HEAD

%prod.quarkus.http.auth.policy.modify-policy.roles-allowed=admin
%prod.quarkus.http.auth.permission.admin.paths=/api/*,/public/*,/css/*,/js/*,/robots.txt
%prod.quarkus.http.auth.permission.admin.policy=modify-policy
%prod.quarkus.http.auth.permission.admin.methods=PUT,POST,DELETE
---------------------------
With this config
the user just can read only and the admin has full permission.